### ISSUE TITLE
the diff of resnet18 between caffe and pytorch model is large

### ISSUE THREAD

**Issue by churitongtong** [xxradon/PytorchToCaffe#16]
I transfer pytorch to caffe of resnet18. 
But the diff is too large.
Input is random, the diff is around 0.17.

How could I get to know where is the error?

fc_blob1                       pytorch_shape: (1000,)              caffe_shape: (1000,)              output_diff: 0.161286

I delete the pad:1 in max_pool1 as it show error: F0429 07:03:48.171044 146212 net.cpp:757] Cannot copy param 0 weights from layer 'fc1'; shape mismatch.  Source param shape is 1000 512 (512000); target param shape is 1000 2048 (2048000). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.

**Comment by churitongtong** [comment]
I change this line by adding ceil_mode=True to keep consistent with caffe. Now It's ok.
        #self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
        self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, ceil_mode=True)

**Comment by XupingZHENG** [comment]
@churitongtong 
Hi, It seems that you successfully run the converted model in caffe
what versions of caffe, pytorch and protobuf do you use?
I can convert the pytorch resnet18 to caffe, but encountered error in network forwarding.
The problem was solved by removing all the batch normalization layers in pytorch.

**Comment by ghgggg** [comment]
I also find that pytorch ‘s conv with stride=2 is different with caffe，when I convert hopenet from pytorch to caffe，its shape  changes from （1，2048，1，1） to （1，2048，2，2） at last pooling layer；if i change the kernel size and stride to 8，it seems good but diff is about 0.18~~~
